I'm getting really strange errors and data is changing after addition…

jillianw opened the pull request:
I think something is going wrong with my debugging challenge file. I've run bin/setup multiple times (it helped at first -- going from no webpage loading to having one with student info appear) but now it looks like it's multiplied some of the data. Each time I've run bin/setup the data has changed significantly. Sorry, I'm just catching up with this -- had to come home and have a nap. Sick today...  :mask: 

![image](https://cloud.githubusercontent.com/assets/16272007/12897430/ecda247e-ce6c-11e5-85af-c1e72aaa7628.png)

Last rackup looked like this after refreshing the web page:

[2016-02-08 13:55:15] INFO  WEBrick 1.3.1
[2016-02-08 13:55:15] INFO  ruby 2.3.0 (2015-12-25) [x86_64-darwin15]
[2016-02-08 13:55:15] INFO  WEBrick::HTTPServer#start: pid=13079 port=9292
WARN: tilt autoloading 'tilt/erb' in a non thread-safe way; explicit require 'tilt/erb' suggested.
D, [2016-02-08T13:55:19.035760 #13079] DEBUG -- :   Student Load (0.6ms)  SELECT "students".\* FROM "students"
D, [2016-02-08T13:55:19.087942 #13079] DEBUG -- :    (0.8ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]
D, [2016-02-08T13:55:19.089050 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.095282 #13079] DEBUG -- :    (0.6ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]
D, [2016-02-08T13:55:19.096213 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.097945 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 2]]
D, [2016-02-08T13:55:19.098777 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.100837 #13079] DEBUG -- :    (0.5ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 2]]
D, [2016-02-08T13:55:19.101657 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.103260 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 3]]
D, [2016-02-08T13:55:19.103963 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.105369 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 3]]
D, [2016-02-08T13:55:19.106049 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.107589 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 4]]
D, [2016-02-08T13:55:19.108248 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.109664 #13079] DEBUG -- :    (0.4ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 4]]
D, [2016-02-08T13:55:19.110469 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.112042 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 5]]
D, [2016-02-08T13:55:19.112721 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.114190 #13079] DEBUG -- :    (0.4ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 5]]
D, [2016-02-08T13:55:19.114905 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.116402 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 6]]
D, [2016-02-08T13:55:19.117132 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.118522 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 6]]
D, [2016-02-08T13:55:19.119259 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.120813 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 7]]
D, [2016-02-08T13:55:19.121481 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.123031 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 7]]
D, [2016-02-08T13:55:19.123938 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.127165 #13079] DEBUG -- :    (0.6ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 8]]
D, [2016-02-08T13:55:19.129161 #13079] DEBUG -- :    (0.9ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.131864 #13079] DEBUG -- :    (0.5ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 8]]
D, [2016-02-08T13:55:19.134671 #13079] DEBUG -- :    (1.0ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.137266 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 9]]
D, [2016-02-08T13:55:19.141568 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.143191 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 9]]
D, [2016-02-08T13:55:19.143966 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.145537 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 10]]
D, [2016-02-08T13:55:19.146254 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.147825 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 10]]
D, [2016-02-08T13:55:19.149119 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.151589 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 11]]
D, [2016-02-08T13:55:19.152569 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.154179 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 11]]
D, [2016-02-08T13:55:19.154959 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.156474 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 12]]
D, [2016-02-08T13:55:19.157185 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.158527 #13079] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 12]]
D, [2016-02-08T13:55:19.159560 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.161350 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 13]]
D, [2016-02-08T13:55:19.162121 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.163513 #13079] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 13]]
D, [2016-02-08T13:55:19.164273 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.166292 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 14]]
D, [2016-02-08T13:55:19.167298 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.168909 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 14]]
D, [2016-02-08T13:55:19.169664 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.171164 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 15]]
D, [2016-02-08T13:55:19.171874 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.173281 #13079] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 15]]
D, [2016-02-08T13:55:19.174355 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.176473 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 16]]
D, [2016-02-08T13:55:19.177560 #13079] DEBUG -- :    (0.5ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.179262 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 16]]
D, [2016-02-08T13:55:19.180122 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.182562 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 17]]
D, [2016-02-08T13:55:19.183862 #13079] DEBUG -- :    (0.6ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.185745 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 17]]
D, [2016-02-08T13:55:19.190068 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.191812 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 18]]
D, [2016-02-08T13:55:19.192612 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.194332 #13079] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 18]]
D, [2016-02-08T13:55:19.195170 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.196883 #13079] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 19]]
D, [2016-02-08T13:55:19.197673 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.199514 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 19]]
D, [2016-02-08T13:55:19.200671 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.202697 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 20]]
D, [2016-02-08T13:55:19.203552 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.209310 #13079] DEBUG -- :    (0.6ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 20]]
D, [2016-02-08T13:55:19.212016 #13079] DEBUG -- :    (0.6ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.214896 #13079] DEBUG -- :    (1.0ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 21]]
D, [2016-02-08T13:55:19.216443 #13079] DEBUG -- :    (0.6ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.219156 #13079] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 21]]
D, [2016-02-08T13:55:19.220068 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.222004 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 22]]
D, [2016-02-08T13:55:19.222865 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.227280 #13079] DEBUG -- :    (0.4ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 22]]
D, [2016-02-08T13:55:19.228292 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.230309 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 23]]
D, [2016-02-08T13:55:19.231593 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.240968 #13079] DEBUG -- :    (7.7ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 23]]
D, [2016-02-08T13:55:19.242378 #13079] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.244195 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 24]]
D, [2016-02-08T13:55:19.244920 #13079] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.248288 #13079] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 24]]
D, [2016-02-08T13:55:19.250189 #13079] DEBUG -- :    (0.5ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T13:55:19.251178 #13079] DEBUG -- :   Assignment Load (0.4ms)  SELECT "assignments".\* FROM "assignments"
::1 - - [08/Feb/2016:13:55:19 -0600] "GET / HTTP/1.1" 200 7523 0.2813
^C[2016-02-08 13:56:11] INFO  going to shutdown ...
[2016-02-08 13:56:11] INFO  WEBrick::HTTPServer#start done.

jillianw committed:
I'm getting really strange errors and data is changing after additional bin/setup attempts

jillianw commented:
Am I still having technical issues with this assignment? I feel like I should be getting different error messages than the ones I'm seeing. If I run rake/bundle exec rake I get this (or have we completely moved away from rake?):

➜  web-toy__debugging-challenge git:(master) ✗ rake
rake aborted!
LoadError: cannot load such file -- sinatra/activerecord/rake
/Users/Jillian/Code/web-toy__debugging-challenge/Rakefile:3:in `<top (required)>'
(See full trace by running task with --trace)

➜  web-toy__debugging-challenge git:(master) ✗ bundle exec rake
rake aborted!
Don't know how to build task 'default'

(See full trace by running task with --trace)

And when I run rackup and refresh the page I get this whole crazy mess which doesn't look very helpful to me as far as error messages go:

➜  web-toy__debugging-challenge git:(master) ✗ rackup
[2016-02-08 14:58:27] INFO  WEBrick 1.3.1
[2016-02-08 14:58:27] INFO  ruby 2.3.0 (2015-12-25) [x86_64-darwin15]
[2016-02-08 14:58:27] INFO  WEBrick::HTTPServer#start: pid=13552 port=9292
WARN: tilt autoloading 'tilt/erb' in a non thread-safe way; explicit require 'tilt/erb' suggested.
D, [2016-02-08T14:58:56.677004 #13552] DEBUG -- :   Student Load (0.5ms)  SELECT "students".\* FROM "students"
D, [2016-02-08T14:58:56.734282 #13552] DEBUG -- :    (1.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]
D, [2016-02-08T14:58:56.735551 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.742162 #13552] DEBUG -- :    (0.5ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]
D, [2016-02-08T14:58:56.743105 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.744809 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 2]]
D, [2016-02-08T14:58:56.745702 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.747084 #13552] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 2]]
D, [2016-02-08T14:58:56.747752 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.749206 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 3]]
D, [2016-02-08T14:58:56.749893 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.751318 #13552] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 3]]
D, [2016-02-08T14:58:56.752140 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.753769 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 4]]
D, [2016-02-08T14:58:56.754446 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.755820 #13552] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 4]]
D, [2016-02-08T14:58:56.756598 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.758653 #13552] DEBUG -- :    (0.5ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 5]]
D, [2016-02-08T14:58:56.759572 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.761455 #13552] DEBUG -- :    (0.5ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 5]]
D, [2016-02-08T14:58:56.762343 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.763903 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 6]]
D, [2016-02-08T14:58:56.764583 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.766015 #13552] DEBUG -- :    (0.3ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 6]]
D, [2016-02-08T14:58:56.766722 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:58:56.767609 #13552] DEBUG -- :   Assignment Load (0.3ms)  SELECT "assignments".\* FROM "assignments"
::1 - - [08/Feb/2016:14:58:56 -0600] "GET / HTTP/1.1" 200 2199 0.1545
D, [2016-02-08T14:59:11.811349 #13552] DEBUG -- :   Student Load (0.5ms)  SELECT "students".\* FROM "students"
D, [2016-02-08T14:59:11.814809 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]
D, [2016-02-08T14:59:11.815665 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.817102 #13552] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]
D, [2016-02-08T14:59:11.817796 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.819253 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 2]]
D, [2016-02-08T14:59:11.819960 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.821256 #13552] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 2]]
D, [2016-02-08T14:59:11.821947 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.823409 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 3]]
D, [2016-02-08T14:59:11.824077 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.825445 #13552] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 3]]
D, [2016-02-08T14:59:11.826325 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.827931 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 4]]
D, [2016-02-08T14:59:11.828642 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.830032 #13552] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 4]]
D, [2016-02-08T14:59:11.830772 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.832306 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 5]]
D, [2016-02-08T14:59:11.832992 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.834273 #13552] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 5]]
D, [2016-02-08T14:59:11.835369 #13552] DEBUG -- :    (0.4ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.837193 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 6]]
D, [2016-02-08T14:59:11.837947 #13552] DEBUG -- :    (0.2ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.839357 #13552] DEBUG -- :    (0.2ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 6]]
D, [2016-02-08T14:59:11.840317 #13552] DEBUG -- :    (0.3ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T14:59:11.840900 #13552] DEBUG -- :   Assignment Load (0.2ms)  SELECT "assignments".\* FROM "assignments"
::1 - - [08/Feb/2016:14:59:11 -0600] "GET / HTTP/1.1" 200 2199 0.0369
D, [2016-02-08T14:59:39.260229 #13552] DEBUG -- :   Student Load (0.3ms)  SELECT  "students".\* FROM "students" WHERE "students"."id" = $1 LIMIT 1  [["id", 2]]
D, [2016-02-08T14:59:39.263110 #13552] DEBUG -- :   Assignment Load (0.5ms)  SELECT "assignments".\* FROM "assignments"
D, [2016-02-08T14:59:39.266076 #13552] DEBUG -- :   Submission Load (0.3ms)  SELECT  "submissions".\* FROM "submissions" WHERE "submissions"."student_id" = $1 AND "submissions"."assignment_id" = $2 LIMIT 1  [["student_id", 2], ["assignment_id", 1]]
D, [2016-02-08T14:59:39.282342 #13552] DEBUG -- :   Grade Load (0.3ms)  SELECT  "grades".\* FROM "grades" WHERE "grades"."submission_id" = $1 LIMIT 1  [["submission_id", 2]]
D, [2016-02-08T14:59:39.288355 #13552] DEBUG -- :   Submission Load (0.4ms)  SELECT  "submissions".\* FROM "submissions" WHERE "submissions"."student_id" = $1 AND "submissions"."assignment_id" = $2 LIMIT 1  [["student_id", 2], ["assignment_id", 2]]
D, [2016-02-08T14:59:39.289466 #13552] DEBUG -- :   Grade Load (0.2ms)  SELECT  "grades".\* FROM "grades" WHERE "grades"."submission_id" = $1 LIMIT 1  [["submission_id", 3]]
D, [2016-02-08T14:59:39.290566 #13552] DEBUG -- :   Submission Load (0.2ms)  SELECT  "submissions".\* FROM "submissions" WHERE "submissions"."student_id" = $1 AND "submissions"."assignment_id" = $2 LIMIT 1  [["student_id", 2], ["assignment_id", 3]]
D, [2016-02-08T14:59:39.291428 #13552] DEBUG -- :   Submission Load (0.2ms)  SELECT  "submissions".\* FROM "submissions" WHERE "submissions"."student_id" = $1 AND "submissions"."assignment_id" = $2 LIMIT 1  [["student_id", 2], ["assignment_id", 4]]
D, [2016-02-08T14:59:39.292302 #13552] DEBUG -- :   Submission Load (0.2ms)  SELECT  "submissions".\* FROM "submissions" WHERE "submissions"."student_id" = $1 AND "submissions"."assignment_id" = $2 LIMIT 1  [["student_id", 2], ["assignment_id", 5]]
::1 - - [08/Feb/2016:14:59:39 -0600] "GET /students/2 HTTP/1.1" 200 1592 0.0431
::1 - - [08/Feb/2016:14:59:40 -0600] "GET /assignments/2 HTTP/1.1" 404 515 0.0022
^C[2016-02-08 15:00:54] INFO  going to shutdown ...
[2016-02-08 15:00:54] INFO  WEBrick::HTTPServer#start done.

sumeetjain commented:
No need to run `rake` commands here. Just `rackup`. That seems to be working just fine.

sumeetjain commented:
Try going to a student's profile or other pages. See what errors that causes.

jillianw commented:
Student pages show up, assignment pages do not. They say Sinatra doesn't know this ditty... and give me this:

```
# in main.rb
class MyApp
  get '/assignment/1' do
    "Hello World"
  end
end
```

Which does not make sense to me. There's no "main.rb" file. What's "Hello World" got to do with it? Am I missing something?

jillianw commented:
Could I please get some guidance on the error copied below? I can't get past it. Worksheet says 
1. Read the error carefully and thoughtfully to see if there is meaningful information in it.
2. Locate the error in code, starting by going to the file and line number identified in the error.

I am not seeing meaningful information in it. It is not giving me one of the error types listed nor is it telling me a file or a line to go to. 

D, [2016-02-08T15:40:02.302034 #13784] DEBUG -- :   Student Load (0.5ms)  SELECT "students".\* FROM "students"
D, [2016-02-08T15:40:02.360262 #13784] DEBUG -- :    (0.8ms)  SELECT COUNT(_) FROM "assignments" INNER JOIN "submissions" ON "assignments"."id" = "submissions"."assignment_id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]
D, [2016-02-08T15:40:02.361845 #13784] DEBUG -- :    (0.5ms)  SELECT COUNT(_) FROM "assignments"
D, [2016-02-08T15:40:02.369167 #13784] DEBUG -- :    (0.5ms)  SELECT SUM("grades"."score") FROM "grades" INNER JOIN "submissions" ON "grades"."submission_id" = "submissions"."id" WHERE "submissions"."student_id" = $1  [["student_id", 1]]

sumeetjain commented:
None of what you pasted is an error. Are you working on this: http://cl.ly/f5L1/Screen%20Shot%202016-02-08%20at%203.59.33%20PM.png ? If so, I can help with that.

Otherwise, I can help you figure out how to move forward to a new error.

jillianw commented:
If none of that is an error, how do I get it to show me what errors there are? Just rackup right? That's what I'm doing. And it's giving me bunches of lines of those non-errors. That's it.

Yes, I would work on the http://cl.ly/f5L1/Screen%20Shot%202016-02-08%20at%203.59.33%20PM.png if that made any sense either. I'm lost.

sumeetjain commented:
Gotcha, I think I see the miscommunication.

First, there is no list of errors. There is only a list of _features_ that should work: https://github.com/omahacodeschool/web-toy__debugging-challenge#homepage

You should try yourself to run through each "story". Many of them won't work or will be incomplete.

When you encounter an actual error, that will be the start of a debugging session.

---

The "Sinatra doesn't know this ditty" message is Sinatra telling you, "You're trying to access a request path (like `'/assignment/5'`) that doesn't have a matching controller action anywhere.

Then it says, "Maybe some of this sample code below will help? But I don't know what your code is structured like, so maybe it won't." (It won't. Because, like you said, we don't have a main.rb file.)

sumeetjain commented:
The lines in the logs that have `DEBUG` in them are not error lines. You can ignore them completely. Look for error lines that look something like what we went over in class today.

sumeetjain commented:
For the error that you get when trying to load an assignment page, that _is_ one of the errors I intend for you to figure out a solution to.

Consider a request to "/about": What happens once the request is made? How does Sinatra figure out what Ruby to run and what view file to send back as the response?

Which of those steps that happens after the request is made might be failing in this case?

jillianw committed:
Added overall_grade_adjusted to home.erb and now have a NoMethodError showing instead of homepage.

jillianw committed:
homepage working again

jillianw committed:
Homepage broken again after I made a syntax correction to admin controller (view_assignment)

jillianw committed:
fixed it -- syntax error in home.erb (@studentS)

jillianw committed:
Finally getting a REAL error page for assignment links.

jillianw committed:
Absolutely can't figure out why the assigment pages are still showing an error page. Looked at other peoples' finished code for this specific line and I'm not seeing any differences there. Error:

NoMethodError at /assignments/5
undefined method `title' for nil:NilClass
file: view_assignment.erb location: block in singleton class line: 3

jillianw committed:
done
